util/metrics/src/lib.rs

```diff
@@ -1,81 +1,170 @@
+#![allow(missing_docs)]
 //! A lightweight metrics facade used in CKB.
 //!
 //! The `ckb-metrics` crate is a set of tools for metrics.
 //! The crate [`ckb-metrics-service`] is the runtime which handles the metrics data in CKB.
 //!
 //! [`ckb-metrics-service`]: ../ckb_metrics_service/index.html
-//!
-//! ## Use
-//!
-//! The basic use of the facade crate is through the metrics macro: [`metrics!`].
-//!
-//! ### Examples
-//!
-//! ```rust
-//! use ckb_metrics::{metrics, Timer};
-//!
-//! fn run_query(_: &str) -> u64 { 42 }
-//!
-//! pub fn process(query: &str) -> u64 {
-//!     let timer = Timer::start();
-//!     let row_count = run_query(query);
-//!     metrics!(timing, "process.query_time", timer.stop());
-//!     metrics!(counter, "process.query_row_count", row_count);
-//!     row_count
-//! }
-//! ```
-
-use opentelemetry::metrics::Meter;
-use std::time::{Duration, Instant};
-
-#[doc(hidden)]
-pub use opentelemetry as internal;
-
-/// Returns a global meter.
-pub fn global_meter() -> Meter {
-    opentelemetry::global::meter("ckb-metrics")
-}
 
-/// A simple timer which is used to time how much time elapsed.
-pub struct Timer(Instant);
+use lazy_static::lazy_static;
+use prometheus::{
+    register_int_counter, register_int_gauge, register_int_gauge_vec, IntCounter, IntGauge,
+    IntGaugeVec,
+};
+use prometheus_static_metric::make_static_metric;
+
+lazy_static! {
+    /// CKB Chain Tip Header Number
+    pub static ref CKB_CHAIN_TIP: IntGauge =
+        prometheus::register_int_gauge!("ckb_chain_tip", "The CKB Chain Tip Number").unwrap();
```

Comment:
:+1: for the metric description.

Reply:
I should change the comment to `/// IntGauge metric for CKB Chain Tip Header Number`

Reply:
I updated the metrics's comments and help messages.